ConsumeRequestPayloadOnResponsePathTest.testConsumeRequestPayloadAfterTrailersSent test failure

https://ci.servicetalk.io/job/servicetalk-master-snapshot/378/testReport/junit/io.servicetalk.http.netty/ConsumeRequestPayloadOnResponsePathTest/testConsumeRequestPayloadAfterTrailersSent

```
Error Message

java.lang.AssertionError: 
Expected: is "ExpectedRequestPayload"
     but: was ""
Stacktrace

java.lang.AssertionError: 
Expected: is "ExpectedRequestPayload"
     but: was ""
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
	at org.junit.Assert.assertThat(Assert.java:956)
	at org.junit.Assert.assertThat(Assert.java:923)
	at io.servicetalk.http.netty.ConsumeRequestPayloadOnResponsePathTest.test(ConsumeRequestPayloadOnResponsePathTest.java:174)
	at io.servicetalk.http.netty.ConsumeRequestPayloadOnResponsePathTest.testConsumeRequestPayloadAfterTrailersSent(ConsumeRequestPayloadOnResponsePathTest.java:104)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:168)
	at io.servicetalk.concurrent.internal.ServiceTalkTestTimeout$TimeoutStatement$CallableStatement.call(ServiceTalkTestTimeout.java:160)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:748)
```

--- a/servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ConsumeRequestPayloadOnResponsePathTest.java
+++ b/servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/ConsumeRequestPayloadOnResponsePathTest.java
@@ -15,7 +15,6 @@
  */
 package io.servicetalk.http.netty;
 
-import io.servicetalk.buffer.api.CompositeBuffer;
 import io.servicetalk.concurrent.api.Completable;
 import io.servicetalk.concurrent.api.Single;
 import io.servicetalk.concurrent.internal.PlatformDependent;
@@ -64,7 +63,7 @@ public class ConsumeRequestPayloadOnResponsePathTest {
 
     private final CountDownLatch waitServer = new CountDownLatch(1);
     private final AtomicReference<Throwable> errorRef = new AtomicReference<>();
-    private final CompositeBuffer receivedPayload = DEFAULT_ALLOCATOR.newCompositeBuffer();
+    private final StringBuffer receivedPayload = new StringBuffer();
 
     @Test
     public void testConsumeRequestPayloadBeforeResponseMetaDataSent() throws Exception {
@@ -135,7 +134,7 @@ public class ConsumeRequestPayloadOnResponsePathTest {
     // TODO: add testTrailersSentAndConsumeRequestPayload when Publisher.merge(Completable) is available
 
     private Completable consumePayloadBody(final StreamingHttpRequest request) {
-        return request.payloadBody().beforeOnNext(receivedPayload::addBuffer)
+        return request.payloadBody().beforeOnNext(buffer -> receivedPayload.append(buffer.toString(UTF_8)))
                 .ignoreElements()
                 .beforeOnError(errorRef::set)
                 .afterFinally(waitServer::countDown);
@@ -174,7 +173,7 @@ public class ConsumeRequestPayloadOnResponsePathTest {
             waitServer.await();
             assertThat(response.status(), is(OK));
             assertThat("Request payload body might be consumed by someone else", errorRef.get(), is(nullValue()));
-            assertThat(receivedPayload.toString(UTF_8), is(EXPECTED_REQUEST_PAYLOAD));
+            assertThat(receivedPayload.toString(), is(EXPECTED_REQUEST_PAYLOAD));
             assertThat(response.headers().contains(TRAILER, X_TOTAL_LENGTH), is(true));
             assertThat(response.trailers().contains(X_TOTAL_LENGTH), is(true));
             CharSequence trailerLength = response.trailers().get(X_TOTAL_LENGTH);

